### ISSUE TITLE
Add regularization to the loss function

### ISSUE THREAD

**Issue by manuelli** [RobotLocomotion/pytorch-dense-correspondence#85]
Currently we have no weight regularization in the loss function. We do have weight decay in the Adam optimizer however.
